_posts/2019-08-09-iglu-scala-client-0.6.0-released/index.md

```diff
@@ -0,0 +1,116 @@
+---
+layout: post
+title: Iglu Scala Client 0.6.0 released
+title-short: Iglu Scala Client
+tags: [iglu, scala, json, jsonschema]
+author: Anton
+category: Releases
+permalink: /blog/2019/08/09/iglu-scala-client-0.6.0-released/
+---
+
+We're tremendously excited to announce the new 0.6.0 release of the Iglu Scala Client, a library in charge of schema resolution and data validation in all Snowplow components, including enrichment jobs and loaders. This release brings enormous amount of API changes we've made in order to facilitate implementation of Snowplow Platform Improvement Proposals, including [new bad rows format][bad-rows-rfc], [Amazon Redshift automigrations][automigrations-rfc] and [deprecation of a batch pipeline][batch-deprecation-rfc].
+
+In the rest of this post we will cover:
+
+1. [API Changes](#api-changes)
+2. [Semantic Changes](#semantic-changes)
+3. [New Validator](#new-validator)
+4. [Acknowledgements](#acknowledgements)
+5. [Roadmap and upcoming features](#roadmap)
+6. [Getting help](#help)
+
+<!--more-->
+
+<h3 id="api-changes">1. API changes</h3>
+
+Since 0.6.0 Iglu Scala Client exposes a new class called Client which consists of two independent entities: Resolver and Validator. Resolver is responsible for schema resolution, caching and error handling and Validator receives resolved schema with datum user wants to validate and returns the validation report. These entities can be used separately or even re-defined by user, but it is recommended to use Client class as an abstraction for most common use case - validation of self-describing entities.
+
+The Client class defines only one function:
+
+{% highlight scala %}
+def check[F[_]: RegistryLookup: Clock: Monad, A](instance: SelfDescribingData[A]): EitherT[F, ClientError, Unit]
+{% endhighlight %}
+
+Where:
+
+* F[_] is an abstract effect type, requiring a [tagless final][tagless-final] capabilities RegistryLookup and `cats.effect.Clock` as well as instance of cats.Monad type class. Two most common concrete effect types are cats.effect.IO and Id, all necessary machinery for them provided out-of-box, but user can also define this machinery for ZIO, Monix Task or sophisticated test type
```

Comment:
as well as instance of cats.Monad type class.  The two most common effect types are cats.effect.IO and Id, all necessary machinery for them are provided out-of-box, but users can also define this machinery for ZIO, Monix Task or sophisticated test types.